src/transformers/pipelines/base.py

```diff
@@ -842,7 +842,16 @@ def __init__(
                 # Take the first device used by `accelerate`.
                 device = next(iter(hf_device_map.values()))
             else:
-                device = -1
+                if (
+                    is_torch_mlu_available()
+                    or is_torch_cuda_available()
+                    or is_torch_npu_available()
+                    or is_torch_xpu_available(check_device=True)
+                    or is_torch_mps_available()
+                ):
+                    device = 0
+                else:
+                    device = -1
```

Comment:
because a line was deleted above, I can't make the full code suggestions unfortunately.

Rather than:
```py
else:
    if (...):
        ...
    else: 
        ...
```

we should do

```py
elif (...):
    ...
else: 
    ...
```


```suggestion
            elif (
                is_torch_mlu_available()
                or is_torch_cuda_available()
                or is_torch_npu_available()
                or is_torch_xpu_available(check_device=True)
                or is_torch_mps_available()
            ):
                # Automatically place on device if available in the environment
                device = 0
            else:
                device = -1
```

Reply:
Thx for the correction!